### PR TITLE
Added frametick event, fired upon every frame start.

### PR DESCRIPTION
**Pull request by michielvandergeest**
Allows developers to tap into frame loop

### CODE REVIEW

**Comment by chiefcll** [comment]
I don't think we should be adding so much to animations. Already getting stutter on simple animations. We're starting to add lot of overheard for things most apps won't need

File: src/core/Stage.ts [hunk 1]
```diff
@@ -184,13 +193,19 @@ export class Stage extends EventEmitter {
     if (!this.root) {
       return;
     }
+    this.frameCount++;
```
[the remark this answers is not in the document]
Comment (michielvandergeest) [reply src/core/Stage.ts:196]: Fair point, this did cross my mind as well. ```frameCount``` can indeed easily become a large number. 

In L2 apps I've mostly seen the frameCount used in combination with ``%```, so I'm not sure anyone is ever interested in the actual number itself. Not sure how to keep the number as low as possible though, without losing out on providing necessary info - @erikhaandrikman / @DouweCnossen: any insights on what info would be needed?

[the remark this answers is not in the document]
Comment (erikhaandrikman) [reply src/core/Stage.ts:196]: We indeed should reconsider whether it's necessary to track the total number of frames generated from start, also keeping @wouterlucas concerns into consideration. I'm exploring potential use cases in where developers would actually need the exact amount of frames generated at any given moment. It's probably more practical to provide `startTime` enabling developer to calculate the total elapsed time and manage Application state based of that number.

If, for any reason, they do need the the exact amount of frames, we might need to encourage them to internally track it within the application, with a certain threshold in mind:

pseudo;
```javascript
const MAX_FRAME_COUNT = 1e6;
let frameCount; 

renderer.on('frameTick', ()=>{
  frameCount++;
  if (frameCount >= MAX_FRAME_COUNT) {
     frameCount = 0;
  }
});

**Review by wouterlucas — APPROVED** [review]
LGTM

**Review by erikhaandrikman — APPROVED** [review]

**Pull request merged**